### PR TITLE
DIRECTOR: Add detection for Mysterious Island; add MystIsle xobj

### PR DESCRIPTION
**Pull request by einstein95**

### CODE REVIEW

File: engines/director/lingo/xlibs/mystisle.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	g_lingo->printSTUBWithArglist("MystIsleXObj::m_new", nargs);
+	g_lingo->dropStack(nargs);
+	g_lingo->push(g_lingo->_state->me);
+}
+
+XOBJSTUBNR(MystIsleXObj::m_dispose)
+XOBJSTUB(MystIsleXObj::m_name, "MystIsle")
+XOBJSTUB(MystIsleXObj::m_getLastError, 0)
+XOBJSTUB(MystIsleXObj::m_getSysVersion, 700)  // game checks < 310 (Win) or < 700 (Mac)
+XOBJSTUB(MystIsleXObj::m_getQTVersion, 200)  // game checks < 110 (Win) or < 200 (Mac)
```
Comment (rvanlaar) [inline engines/director/lingo/xlibs/mystisle.cpp:106]: Would it be better to unstub this and implement a check based on machine type is mac or windows?

See also this code for context: https://github.com/scummvm/scummvm/blob/master/engines/director/lingo/lingo-the.cpp#L542

File: engines/director/detection_tables.h [hunk 2]
```diff
@@ -5031,6 +5032,10 @@ static const DirectorGameDescription gameDescriptions[] = {
 	MACDEMO1_l("mysteriousegypt", "Demo", "Egypti demo", "337c94c66b435e075a2d7f94b3cffc49", 483746, Common::FI_FIN, 404),
 	WINDEMO1_l("mysteriousegypt", "Demo", "EGYPTI.EXE",	 "2e8cba31e473bd1567476f57d6aeb4d6", 5574413, Common::FI_FIN, 404),
 
+	// version comes from debug mode (hold Ctrl+Shift when clicking on New Game/Load Game)
+	MACGAME1("mysteriousisland", "v96.04.23", "Mysterious Island", "r:17efee018a660458fae80de4364021ac", 483518, 404),
+	WINGAME1("mysteriousisland", "v96.04.23", "ISLAND.EXE", "t:d8c7d6a10e60da366408d2cee6510f79", 697057, 404),
```
Comment (tag2015) [inline engines/director/detection_tables.h:5037]: Sorry for the annoyance, but this one definitely needs an additional file

File: engines/director/lingo/xlibs/mystisle.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+XOBJSTUB(MystIsleXObj::m_verifyUnlocked, 0)  // 0: unlocked; 1: locked
```
Comment (rvanlaar) [inline engines/director/lingo/xlibs/mystisle.cpp:107]: Does locked and unlocked mean something?

Comment (einstein95) [reply engines/director/lingo/xlibs/mystisle.cpp:107]: Yeah, it's Mac terminology for a drive being read-only or read/write

File: engines/director/detection_tables.h (hunk 2 above)
Comment (einstein95) [reply engines/director/detection_tables.h]: No problem, added in 5d34d14c1aa

File: engines/director/lingo/xlibs/mystisle.cpp (hunk 1 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/mystisle.cpp]: Done

File: engines/director/lingo/xlibs/mystisle.cpp (hunk 3 above)
Comment (einstein95) [reply engines/director/lingo/xlibs/mystisle.cpp]: Added a comment regarding what it's checking and what we return

**Comment by rvanlaar** [comment]
Hi, If it's not too much trouble, please split the changes into one for mysterious island and one for the Virgin one.

Note that when the commit message contains "and" it's almost always better to split it into two commit.
I have no further comments on the code. That looks good.

**Comment by einstein95** [comment]
Right, split the commits. Should be good now

**Comment by bluegr** [comment]
Thanks!

**Pull request merged**